mempool/cat/cache.go

```diff
@@ -0,0 +1,292 @@
+package cat
+
+import (
+	"container/list"
+	"time"
+
+	tmsync "github.com/tendermint/tendermint/libs/sync"
+	"github.com/tendermint/tendermint/types"
+)
+
+// LRUTxCache maintains a thread-safe LRU cache of raw transactions. The cache
+// only stores the hash of the raw transaction.
+type LRUTxCache struct {
+	mtx      tmsync.Mutex
+	size     int
+	cacheMap map[types.TxKey]*list.Element
+	list     *list.List
```

Comment:
If I'm understanding this correctly, the size is static. I'm not sure if we have any standards around prefixes cc @evan-forbes but I'd recommend something like `static` so that it is clear that this field doesn't require acquiring the mutex in order to access it. 

Similarly the list should also have the `static` prefix as it manages its own mutex and doesn't require the `LRUTxCache` mutex for its operations to be threadsafe.

```suggestion
	staticSize int
	staticList *list.List
	
	mtx      tmsync.Mutex
	cacheMap map[types.TxKey]*list.Element
```

Reply:
Yeah you're right. I think the convention is to place the mutex only above the fields that access it. Here this was just copied it from the v1 mempool implementation

Reply:
> Similarly the list should also have the static prefix as it manages its own mutex and doesn't require the LRUTxCache mutex for its operations to be threadsafe.

`list.List` does not have it's own mutex and is managed by the LRUTxCache's mutex so I will leave it as is.